tests/scalability/generate_brac_data/generate-brac-data.js

```diff
@@ -0,0 +1,217 @@
+
+const createDataDoc = (directory, content) => {
+  fs.writeFile(directory, content, err => {
+    if (err) {
+      console.error('CreateDataDoc ' + err);
+      return;
+    }
+  });
+};
+
+const createDataDirectory = (path, directoryName) => {
+  if (!fs.existsSync(path + directoryName)) {
+    fs.mkdirSync(path + directoryName, err => {
+      if (err) {
+        console.error('CreateDataDirectory ' + directoryName + ' - ' + err);
+        return;
+      }
+    });
+  }
+};
+
+const setParents = (firstLevelId, secondLevelId, thirdLevelId) => {
+  return {
+    _id: firstLevelId,
+    parent: {
+      _id: secondLevelId,
+      parent: {
+        _id: thirdLevelId
+      }
+    }
+  };
+};
+
+const pairPlaceTypesRoles = {
+  'district_hospital': 'supervisor',
+  'health_center': 'district_admin',
+  'clinic': null
+};
+
+const pairPlaceTypesNeedsUsers = {
+  'district_hospital': true,
+  'health_center': true,
+  'clinic': false
+};
+
+const pairPlaceTypesPersonSubtype = {
+  'district_hospital': 'manager',
+  'health_center': 'chw',
+  'clinic': 'member_eligible_woman'
+};
+
+const generateHierarchy = (type, placeName, numberOfPersons,
+  parentFirstLevelId, parentSecondLevelId, directParentPlace) => {
+
+  const place = bracPlaceFactory.generateBracPlace(placeName, type,
+    setParents(parentFirstLevelId, parentSecondLevelId));
+
+  const needUser = pairPlaceTypesNeedsUsers[type];
+  let parentPersonThirdLevelId = null;
+  let parentPersonSecondLevelId = null;
+  let parentPersonFirstLevelId = null;
+
+  if (type === 'district_hospital') {
+    parentPersonFirstLevelId = place._id;
+  }
+  if (type === 'health_center') {
+    parentPersonFirstLevelId = place._id;
+    parentPersonSecondLevelId = parentFirstLevelId;
+    const randomIndex = Faker.faker.datatype.number(numberOfManagersPerDistrictHospital - 1);
+    place.supervisor = managers[randomIndex]._id;
+  }
+  if (type === 'clinic') {
+    parentPersonFirstLevelId = place._id;
+    parentPersonSecondLevelId = parentFirstLevelId;
+    parentPersonThirdLevelId = parentSecondLevelId;
+  }
+
+  let isPrimaryContact = true;
+  for (let i = 0; i < numberOfPersons; i++) {
+    let roles = pairPlaceTypesRoles[type];
+    let subtype = pairPlaceTypesPersonSubtype[type];
+    if (type === 'clinic' && !isPrimaryContact) {
+      subtype = 'other';
+    }
+    const person = bracPersonFactory.generateBracPerson(setParents(parentPersonFirstLevelId,
+      parentPersonSecondLevelId, parentPersonThirdLevelId), subtype);
+    if (isPrimaryContact) {
+      place.contact = person;
+      createDataDoc(preconditionDataDirectory + place._id + dataExtension, JSON.stringify(place, {}, 2));
+      if (type === 'district_hospital') {
+        roles = ['national_admin', 'mm-online'];
+      }
+    }
+    createDataDoc(preconditionDataDirectory + person._id + dataExtension, JSON.stringify(person, {}, 2));
+    if (type === 'district_hospital') {
+      managers.push(person);
+    }
+    if (needUser) {
```

Comment:
Do you need a user for every person that is added, or just the primary contact?

Reply:
For every person created at District Hospital and Health Center level.
```
const pairPlaceTypesNeedsUsers = {
  'district_hospital': true,
  'health_center': true,
  'clinic': false
};
```